client/tink_keyset_helper.go

```diff
@@ -42,3 +46,35 @@ func nameOfSupportedTinkKeyTemplates() string {
 	sort.Strings(supportedTemplates)
 	return strings.Join(supportedTemplates, "\n")
 }
+
+// checkTemplateNameAndKnoxIDForTinkKeyset checks whether knox identifier start with "tink:<tink_primitive_short_name>:".
+func checkTemplateNameAndKnoxIDForTinkKeyset(templateName string, knoxIentifier string) error {
+	templateInfo, existed := tinkKeyTemplates[templateName]
+	if !existed {
+		return errors.New("not supported keyset template. See 'knox key-templates'")
+	} else if !strings.HasPrefix(knoxIentifier, templateInfo.knoxIDPrefix) {
+		return errors.New("<key_identifier> must have prefix '" + templateInfo.knoxIDPrefix + "'")
+	}
+	return nil
+}
+
+// createNewTinkKeyset creates a new tink keyset contains a single fresh key from the given tink key templateFunc.
+func createNewTinkKeyset(templateFunc func() *tinkpb.KeyTemplate) []byte {
+	// Creates a keyset handle that contains a single fresh key
+	keysetHandle, err := keyset.NewHandle(templateFunc())
+	if keysetHandle == nil || err != nil {
+		fatalf("cannot get tink keyset handle: %v", err)
+	}
+	return convertTinkKeysetHandleToBytes(keysetHandle)
+}
+
+// convertTinkKeysetHandleToBytes extracts keyset from tink keyset handle and converts it to bytes
+func convertTinkKeysetHandleToBytes(keysetHandle *keyset.Handle) []byte {
+	bytesBuffer := new(bytes.Buffer)
+	writer := keyset.NewBinaryWriter(bytesBuffer)
+	// To write cleartext keyset handle, must use package "insecurecleartextkeyset"
+	if err := insecurecleartextkeyset.Write(keysetHandle, writer); err != nil {
```

Comment:
as above, it's more conventional to have the statement on one line and the error check on the line after it.

Reply:
Thank you so much! I have revised my code according to your comment.